.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,28 @@
+name: Pre-commit
+
+on: [push, pull_request]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
+
+      - name: Set up Python
+        uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5.2.0
+        with:
+          python-version: '3.11'
+          cache: pip
+          cache-dependency-path: |
+            **/requirements*.txt
+            .pre-commit-config.yaml
+
+      - name: Install pre-commit
+        run: |
+          python -m pip install --upgrade pip
+          pip install pre-commit
+
+      - name: Run pre-commit
+        run: pre-commit run --all-files
```

Comment:
I don't think we should run it on all files because this means people will get stopped if someone committed something outside of this hook and now their PR will have to mirror all the fixes.

Reply:
That shouldn't happen, right? I think it's a fair expectation that anyone pushing directly to the repo is using the pre-commit hook. 

If that were to happen anyway, they should get an email that this job failed because it will run any time commits are pushed, as well.

With that said, if you still want it to only apply to the diff instead of the full tree, I'll make the change. Just let me know.

Reply:
sounds good!  will update and comment again when it's done

Reply:
took a bit, but i got it working -- passing run on a test PR here: https://github.com/russellb/llama-stack/actions/runs/11168500406/job/31047043034?pr=1

Reply:
nevermind, it's failing here on this PR ...

Reply:
... ok, working for real this time